clang/tools/scan-build-py/lib/libear/ear.c

```diff
@@ -1924,88 +2014,9 @@ static void bear_report_call(char const *fun, char const *const argv[]) {
   free((void *)cwd);
   pthread_mutex_unlock(&mutex);
 #ifdef SYCLomatic_CUSTOMIZATION
-  if (is_nvcc_or_ld == 1 && flag_object == 1) {
-    ret = 1;
-  } else if (is_nvcc_or_ld == 1) {
-    // object is not given by -o. Need figure out the default output for cmd "gcc
-    // -c xx.c"
-    char *tmp = malloc(PATH_MAX);
-    if (tmp == NULL) {
-      perror("bear: malloc memory fail.");
-      exit(EXIT_FAILURE);
-    }
-    memset(tmp, '\0', PATH_MAX);
-    int idx = 0;
-    int c_option;
-    char ofilename[PATH_MAX];
-    memset(ofilename, '\0', PATH_MAX);
-    int parse_ret = 0;
-    for (size_t it = it_cp; it < argc; ++it) {
-      memcpy(tmp + idx, argv[it], strlen(argv[it]));
-      idx += strlen(argv[it]);
-      tmp[idx] = ' ';
-      idx++;
-    }
-    parse_ret = parse_input_file(tmp, &c_option, ofilename);
-    if (parse_ret == 1 && c_option == 1) {
-      // change the suffix of the ofilename from .c .cpp => .o)
-      int olen = strlen(ofilename);
-      while (olen >= 0 && ofilename[olen] != '.') {
-        olen--;
-      }
-      if (olen == -1) {
-        olen = strlen(ofilename);
-        if (olen >= PATH_MAX - 3) {
-          perror("bear: filename length too long.");
-          exit(EXIT_FAILURE);
-        }
-        ofilename[olen] = '.';
-        ofilename[olen + 1] = 'o';
-        ofilename[olen + 2] = '\0';
-      } else {
-        if (olen > PATH_MAX - 3) {
-          perror("bear: filename length too long.");
-          exit(EXIT_FAILURE);
-        }
-        ofilename[olen + 1] = 'o';
-        ofilename[olen + 2] = '\0';
-      }
-
-      pthread_mutex_lock(&mutex);
-      if (generate_file(ofilename, 0) != 0) {
-        pthread_mutex_unlock(&mutex);
-        exit(EXIT_FAILURE);
-      }
-      pthread_mutex_unlock(&mutex);
-    }
-    free(tmp);
-    ret = 1;
-  }
-
-  // try to replace nvcc or clang++ with intercept-stub,
-  // e.g: "/path/to/clang++ -ccbin ... ",
-  //      "/bin/sh -c "/path/to"/bin/clang++ -ccbin ..."
-  const char *pos = find_intercept_compiler(argv[it_cp], compiler_idx);
-
-  int is_stub_need = 0;
-  is_stub_need = (pos != NULL);
-  if (is_stub_need) {
-    ret = 0; // intercept-stub should continue to run.
-
-    // intercept-stub is used to handle the compiler command like
-    // "/bin/sh -c clang++ -c `echo ./`hello.c", it changes the compiler command
-    // intercepted to
-    // "/bin/sh -c /path/to/libear/intercept-stub -c `echo ./`hello.c", then the
-    // coming command "/path/to/libear/intercept-stub -c ./hello.c" will be run,
-    // and the source file name "hello.c" will be captured by intercept.py.
-    argv[it_cp] =
-        replace_binary_name(argv[it_cp], pos, compiler_idx, compiler_array);
-  }
-
-  if (ret == 1 && it_cp == 0 && !is_ar_staticlib_cmd) {
-    exit(0);
+  if(is_tool_existence) {
+    return 0;
```

Comment:
it's void function,  so why " return 0"?

Reply:
It is not void function, shown as follows:
`
static int bear_report_call(char const *fun, char const *argv[]) {
#else
static void bear_report_call(char const *fun, char const *const argv[]) {
#endif // SYCLomatic_CUSTOMIZATION

`